File: examples/wirespec-gradle-plugin-ktor/src/main/kotlin/community/flock/wirespec/examples/app/todo/TodoRepository.kt
package community.flock.wirespec.examples.app.todo

interface HasTodoRepository {
    val todoRepository: TodoRepository
}

interface TodoRepository {

    fun getAllTodos(): List<Todo>

    fun getTodoById(id: Todo.Id): Todo

    fun saveTodo(todo: Todo): Todo

    fun deleteTodoById(id: Todo.Id): Todo
}

File: examples/scala-zio-2/src/main/scala/example/port/TodoRepository.scala
package example.port

import community.flock.wirespec.generated.model.{TodoDto, TodoId}
import zio.*

trait TodoRepository:
  def findAll(): Task[List[TodoDto]]
  def save(todo: TodoDto): Task[TodoDto]
  def findById(id: TodoId): Task[Option[TodoDto]]
  def delete(id: TodoId): Task[Option[TodoDto]]
